lib/Driver/Driver.cpp

```diff
@@ -2259,6 +2252,20 @@ Job *Driver::buildJobsForAction(Compilation &C, const JobAction *JA,
 
       Output->setAdditionalOutputForType(types::TY_ModuleTrace, filename);
     }
+
+    if (C.getArgs().hasArg(options::OPT_emit_tbd, options::OPT_emit_tbd_path)) {
+      if (OI.CompilerMode != OutputInfo::Mode::SingleCompile) {
+        llvm::outs() << "TBD emission has been disabled, because it requires a "
+                     << "single compiler invocation (-wmo or "
+                     << "-force-single-frontend-invocation).\n";
```

Comment:
Nitpick: this should just say `-whole-module-optimization` and not mention either the short form or `-force-single-frontend-invocation`. (We don't really encourage the latter for users because it's implementation-detail-y.)